### PR TITLE
Viz chapter is ready

### PR DESCRIPTION
**Pull request by RiboRings**

### CODE REVIEW

**Comment by antagomir** [comment]
I suggest that @TuomasBorman checks first and I will then see after that if I have further comments.

**Review by TuomasBorman — CHANGES_REQUESTED** [review]
I think this is good like this already. Just some minor things that would be good to consider at least

**Comment by antagomir** [comment]
Thanks! I will check the next round.

File: 19_visualization_techniques.Rmd [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+
+The packages _ALDEx2_, _ANCOM-BC_, _Maaslin2_ and _fido_, which were described in chapter \@ref(differential-abundance), support a number of visualization techniques that are specific to each. The **MA**, also called **Bland-Altman**, and the **MW** plots are characteristic of _ALDEx2_ and show difference in abundance among strains as a function of log-ratio abundance and dispersion, respectively. They both present differentially abundant taxa as red dots, uniformly abundant ones as gray dots and rarely abundant ones as black dots. In the case of _ANCOM-BC_ and _Maaslin2_, the analysis returns several **box plots** of the top differentially abundant experimental and control samples. Finally, the tree-based model produced by _fido_ measures the effect of each parameter in terms of its predicted influence on the variance of the selected variable.
+
+|        | _ALDEx2_         | _ANCOM-BC_  | _Maaslin2_   | _fido_                |
+|:------:|:----------------:|:-----------:|:------------:|:---------------------:|
+| method | `aldex.plot`     | `ancombc`   | `Maaslin2`   | `pibble`, `plot`      |
+| plots  | MA and MW plot   | box plot    | box plot     | effect plot           |
+
```
[the remark this answers is not in the document]
Comment (RiboRings) [reply 19_visualization_techniques.Rmd:599]: I wanted to give a short summary on the visual methods of the chapter on differential abundance, without repeating the examples that are already in that chapter.
Should I just delete this part?

[the remark this answers is not in the document]
Comment (antagomir) [reply 19_visualization_techniques.Rmd:599]: I would remove this. You can store it in misc.Rmd that is in the repository, but not included in the book. We can then use the text if it is useful later

File: 19_visualization_techniques.Rmd [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+rowData(tse_order)$Order <- as.character(order_renamed)
+# plot composition as a bar plot
+plotAbundance(tse_order,
+              abund_values = "relabundance",
+              rank = "Order",
+              order_rank_by = "abund",
+              order_sample_by = "Clostridiales")
+```
+
+For more sophisticated visualizations than those produced with `plotAbundance` and _ggplot2_, the packages _pheatmap_ and _sechm_ provide methods to include feature and sample clusters in a heat map, along with further functionality.
```
[the remark this answers is not in the document]
Comment (antagomir) [reply 19_visualization_techniques.Rmd]: "heatmap" it be

File: 19_visualization_techniques.Rmd [hunk 3]
```diff
@@ -0,0 +1,596 @@
...
+  plot_layout(guides = "collect")
+```
+
+For similar examples, readers are referred to chapter \@ref(community-similarity).
+Further material on the graphic capabilities of _patchwork_ is available in its
+[official package tutorial](https://patchwork.data-imaginist.com/articles/patchwork.html).
+
+## Statistical analysis
+
+### Heat maps
```
Comment (TuomasBorman) [inline 19_visualization_techniques.Rmd:243]: One more 'heat map'

File: 19_visualization_techniques.Rmd [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+  sample_types_annotation +
+  heatmap +
+    plot_layout(design = design, guides = "collect", # Specify layout, collect legends
+                widths = c(0.2, 0.15, 1, 1, 1),
+                heights = c(0.1, 0.15, 0.15, 0.25, 1, 1))
+
+plot
+```
+
+Heat maps find several other applications in biclustering and multi-assay
```
Comment (TuomasBorman) [inline 19_visualization_techniques.Rmd:587]: 'Heat map'

**Review by TuomasBorman — APPROVED** [review]
Looks good!

**Comment by TuomasBorman** [comment]
One more suggestion:

We have here` pheatmap` and sechm. `sechm` utilizes `complexHeatmap`. It is like better version of `pheatmap` with more options. 

`pheatmap` is valid choice for plotting matrices, however, since we introduce `sechm` (and `complexHeatmap`), should we have a small example about `complexHeatmap` + matrix? (Then user could use only `complexHeatmap`/`sechm`)

In my opinion, `complexHeatmap` is little bit more complicated but allows much more. (For example, adding text such as p-values is little bit harder than in `pheatmap`)

```
p_threshold_vis <- 0.05

# Create a heatmap and store it
plot <- Heatmap(correlations$cor, col = col_fun,
                # Print values to cells
                cell_fun = function(j, i, x, y, width, height, fill) {
                    # If the p-value is under threshold
                    if( correlations$p_adj[i, j] < p_threshold_vis )
                        # Print "X"
                        grid.text(sprintf("%s", "X"), x, y, gp = gpar(fontsize = 8, col = "black"))
                    },
                heatmap_legend_param = list(title = "", legend_height = unit(5, "cm"))
                )
```

**Comment by antagomir** [comment]
@RiboRings can you resolve the conflict, and confirm that all the relevant points from the above discussion have been taken into account. Then I can confirm the merge. Thanks!

**Comment by antagomir** [comment]
Same than with #134  - conflict must be resolved before we can merge this.

**Comment by antagomir** [comment]
There still seems to be a build failure - we can discuss in Slack how to solve if not straightfwd. @RiboRings

**Review by antagomir — APPROVED** [review]

**Comment by antagomir** [comment]
Still some fail checks, I am not sure from the logs what's the issue. Do you find?

**Comment by RiboRings** [comment]
I'm not sure yet.

**Comment by antagomir** [comment]
Ah - I think I have added this in another recent PR. Have you made sure your branch is updated from master?

**Comment by RiboRings** [comment]
The error about sechm dep was solved. However, now another error appears in the logs. It is about [this chunk](https://github.com/microbiome/OMA/blob/e02dfe32589ce8eb65ea4f95c0ddee65a5d6095b/19_visualization_techniques.Rmd#L422):

```
Quitting from lines 423-436 (19_visualization_techniques.Rmd) 
Error in is.vector(features) : argument "features" is missing, with no default
```

Locally, I don't get any error. Do you have any idea what could cause this error?

**Comment by antagomir** [comment]
The new version of the _sechm_ package is using argument name "features" instead of "genes". Change that on line 430 and possibly elsewhere, then it probably goes through. We had this issue also in some other Rmd files.

**Pull request merged**